test/e2e/clusterctl_upgrade.go

```diff
@@ -746,7 +746,9 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 		// Dumps all the resources in the spec namespace, then cleanups the cluster object and the spec namespace itself.
 		if input.UseKindForManagementCluster {
 			managementClusterProxy.Dispose(ctx)
-			managementClusterProvider.Dispose(ctx)
+			if !input.SkipCleanup {
```

Comment:
Should we also skip the line above? Seems helpful to also keep the kubeconfig

Reply:
Good point. Also then matches what we do normally at the teardown function.